### ISSUE TITLE
Assigment of customer groups is broken

### ISSUE THREAD

**Issue by ChristianP85** [PyVRP/PyVRP#681]
In version 0.10.0 I get an error when I change the order of the group assignment

**Code sample**

```python
from pyvrp import Model
from pyvrp.show_versions import show_versions

show_versions()

m = Model()

A_group = m.add_client_group(False)
B_group = m.add_client_group(False)

depot = m.add_depot(x=0, y=0)

# If I change the order of the two lines it works again
B = m.add_client(x=0, y=0, required=False, group=B_group) 
A = m.add_client(x=0, y=0, required=False, group=A_group)

print("Clients in Groups")
for x in m.groups:
    print(x.clients)

```
**Output (Both Clients are in one group --> ValueError: Empty client group not understood.):**
```
INSTALLED VERSIONS
------------------
     pyvrp: 0.10.0
     numpy: 2.1.2
matplotlib: 3.9.2
    vrplib: 1.5.1
      tqdm: 4.66.5
     tomli: 2.0.2
    Python: 3.11.10
Clients in Groups
[1, 2]
[]
```
**Expected Output (Clients are in different groups):**
```
INSTALLED VERSIONS
------------------
     pyvrp: 0.10.0
     numpy: 2.1.2
matplotlib: 3.9.2
    vrplib: 1.5.1
      tqdm: 4.66.5
     tomli: 2.0.2
    Python: 3.11.10
Clients in Groups
[1]
[2]
```
**Note**: 
In version 0.9.1 the order did not matter and the group assignment was correct

**Comment by N-Wouda** [comment]
Hi @ChristianP85, thanks for reporting the issue! I can reproduce it and have identified the cause: we check these groups by equality rather than ID, now that we implemented our own `__eq__` methods in #646.

I will push a fix probably tomorrow, and then release 0.10.1 to fix this.

**Comment by ChristianP85** [comment]
Wow, thank you! Many thanks for the quick support and the great work! I will try it out tomorrow.
